2018/2/solution2.py

```diff
@@ -0,0 +1,28 @@
+def get_checksum(box_ids):
+    return (len([box_id for box_id in box_ids if contains_matching(2, box_id)])
+            * len([box_id for box_id in box_ids if contains_matching(3, box_id)]))
+
+
+def contains_matching(n, box_id):
+    letters = {c: box_id.count(c) for c in set(c for c in box_id)}
+    return any([True for letter in letters if letters[letter] == n])
+
+
+def get_common_letters(box_ids):
+    for i in range(0, len(box_ids)):
```

Comment:
There's a fair bit of nesting here, but it's still pretty readable?  Do you have any ideas how to lessen the nesting while preserving readability?  The only idea I have is to extract the two for loops to a generator  or use something in itertools, but I'm not sure that's more readable

Reply:
I would use combinations, like you mentioned earlier. I would also turn this into a list comprehension and call next() on it.